app/routes.py

```diff
@@ -1,2 +1,296 @@
-from flask import Blueprint
+import requests
+import os
+from flask import request, Blueprint, make_response, jsonify
 
+from app import db
+from app.models.customer import Customer
+from app.models.video import Video
+from app.models.rental import Rental
+from datetime import datetime, timedelta 
+
+# <------------------------------------- CUSTOMER ENDPOINTS ------------------------------------->
+
+customers_bp = Blueprint("customers", __name__, url_prefix= "/customers")
+
+# endpoint retrieves details about ALL customers and creates a record of a customer
+@customers_bp.route("", methods=["GET", "POST"], strict_slashes=False)
+def customer_details():
+
+    # GET all customer details 
+    
+    if request.method == "GET":
+
+        all_customers = Customer.query.all()
+
+        customers_response = []
+        for customer in all_customers:
+            customers_response.append({
+                "id": customer.id,
+                "name": customer.name,
+                "registered_at": customer.registered_at,
+                "postal_code": customer.postal_code,
+                "phone": customer.phone,
+                "videos_checked_out_count": customer.videos_checked_out_count
+            })
+        return jsonify(customers_response), 200
+    
+    # CREATES (POST) a new customer
+
+    elif request.method == "POST":
+
+        request_body = request.get_json()
+        if "name" not in request_body or "postal_code" not in request_body or "phone" not in request_body:
+            return jsonify({"details": "Invalid data"}), 400
+        else:
+            new_customer = Customer(
+                name=request_body["name"], 
+                postal_code=request_body["postal_code"], 
+                phone=request_body["phone"],
+                registered_at=datetime.now()
+                )
+
+            db.session.add(new_customer)
+            db.session.commit()
+
+            return new_customer.to_json(), 201
+        
+# endpoint retrieves, updates and deletes records of a specific customer
+@customers_bp.route("/<customer_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def specific_customer(customer_id):
+
+    # GET details about specific customer
+
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return make_response(), 404
```

Comment:
We could use `get_or_404()` method here 
https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/#flask_sqlalchemy.BaseQuery.get_or_404